src/main/java/net/querz/nbt/tag/CompoundTag.java

```diff
@@ -10,14 +8,21 @@
 import java.util.Set;
 import java.util.function.BiConsumer;
 
-public class CompoundTag extends Tag<Map<String, Tag<?>>> implements Iterable<Map.Entry<String, Tag<?>>>, Comparable<CompoundTag>, MaxDepthIO {
+import net.querz.io.MaxDepthIO;
+
+public class CompoundTag extends Tag<Map<String, Tag<?>>>
+		implements Iterable<Map.Entry<String, Tag<?>>>, Comparable<CompoundTag>, MaxDepthIO {
 
 	public static final byte ID = 10;
 
 	public CompoundTag() {
 		super(createEmptyValue());
 	}
 
+	public CompoundTag(int initialCapacity) {
```

Comment:
Would it make sense to change `clone()` as well so it sizes the new `CompoundTag` accordingly?
(Though I am not sure if using `size()` as initial capacity would suffice or if spare space should be reserved (`size() + x`) to prevent immediate rehashing when new entries are added. However, whether or not that is acceptable probably depends on the use case.)

Reply:
In our use case we multiply our desired size by 0*75 to count that the hashmap resizes it once its 75% full (load factor param). Wanted to keep it with the same naming and let the caller decide if they want to do something similar 👀

Reply:
As for setting the same size in the clone, I jus pushed a change that avoids resizing on those too!

Reply:
Ah right, the initial capacity of `HashMap` is affected by the load factor as well, so you have adjust it to prevent rehashing.
However for `clone()` `size() * 1.75f` is larger than necessary to fit all existing entries, `size() / 0.75f` (≈ `size() * 1.33`) would suffice, is that intended?

Additionally, even though for `ArrayList` the resizing is an implementation detail, it appears it currently only happens when there is no free space left, so using `size() * 1.75f` for `ListTag` might make the backing ArrayList too large.

Reply:
Did the math wrong haha. Numbers now fixed (listtag uses same size, hashmap uses size() *1.33, ceiled up)

Reply:
Maybe it would be good to use `size() / 0.75f` though and add a comment that this is the default load factor to make it more clear where this value is coming from and why it is used. (I only mentioned `1.33` above to show that it is not the same as `1.75`).

Reply:
if we use size / 0.75 it would resize right?